packages/new-deploy/src/DeployContext.ts

```diff
@@ -0,0 +1,225 @@
+    config.validateConfig(this.data);
+  }
+
+  getDeployer(nameOrDomain: string | number): ethers.Signer {
+    return this.provider.mustGetSigner(nameOrDomain);
+  }
+
+  getDomainConfig(nameOrDomain: string | number): config.Domain | undefined {
+    const name = this.provider.mustResolveDomainName(nameOrDomain);
+    return this.data.protocol.networks[name];
+  }
+
+  mustGetDomainConfig(nameOrDomain: string | number): config.Domain {
+    const protocol = this.getDomainConfig(nameOrDomain);
+    if (!protocol)
+      throw new Error(`No protocol block for domain ${nameOrDomain}`);
+    return protocol;
+  }
+
+  getCore(nameOrDomain: string | number): CoreContracts | undefined {
+    const name = this.provider.mustResolveDomainName(nameOrDomain);
+    if (!this.data.core[name]) return undefined;
+    return new CoreContracts(this, name, this.data.core[name]);
+  }
+
+  mustGetCore(nameOrDomain: string | number): CoreContracts {
+    const core = this.getCore(nameOrDomain);
+    if (!core) throw new Error(`No core contracts for domain ${nameOrDomain}`);
+    return core;
+  }
+
+  getBridge(nameOrDomain: string | number): BridgeContracts | undefined {
+    const name = this.provider.mustResolveDomainName(nameOrDomain);
+    if (!this.data.bridge[name]) return undefined;
+    return new BridgeContracts(this, name, this.data.bridge[name]);
+  }
+
+  mustGetBridge(nameOrDomain: string | number): BridgeContracts {
+    const bridge = this.getBridge(nameOrDomain);
+    if (!bridge)
+      throw new Error(`No bridge contracts for domain ${nameOrDomain}`);
+    return bridge;
+  }
+
+  addDomain(domain: config.Domain): void {
+    this._data = config.addNetwork(this.data, domain);
+  }
+
+  protected addCore(name: string, core: config.EvmCoreContracts): void {
+    this._data = config.addCore(this.data, name, core);
+  }
+
+  protected addBridge(name: string, bridge: config.EvmBridgeContracts): void {
+    this._data = config.addBridge(this.data, name, bridge);
+  }
+
+  pushVerification(
+    nameOrDomain: string | number,
+    verification: Verification,
+  ): void {
+    const name = this.provider.mustResolveDomainName(nameOrDomain);
+    if (!this.verification.has(name)) this.verification.set(name, []);
+
+    const net = this.verification.get(name);
+    if (net) net.push(verification);
+  }
+
+  async deployCore(domain: config.Domain): Promise<void> {
+    this.addDomain(domain);
+
+    const core = new CoreContracts(this, domain.name);
+    await core.recordStartBlock();
+
+    await Promise.all([
+      core.deployUpgradeBeaconController(),
+      core.deployUpdaterManager(),
+      core.deployXAppConnectionManager(),
+    ]);
+
+    await Promise.all([core.deployHome(), core.deployGovernanceRouter()]);
+
+    // all contracts deployed
+    const complete = core.complete();
+    this.addCore(domain.name, complete);
+
+    // the set of domains that are not the new core
+    const remoteDomains = this.data.networks.filter(
```

Comment:
not exactly. We enroll all governance routers and watchers _even if they do not have a connection_ so that if an additional connections is made between those domains, it will already have the correct configuration waiting for it. essentially we're amortizing potential future governance transactions right now


I'll leave a comment to this effect in the code

Reply:
SGTM